hf_xet/src/lib.rs

```diff
@@ -106,53 +141,109 @@ fn try_parse_progress_updaters(funcs: Vec<Py<PyAny>>) -> PyResult<Vec<Arc<dyn Pr
     Ok(updaters)
 }
 
+// TODO: on the next major version update, remove this type alias
+// This is used to support backward compatibility for PyPointerFile with old versions of huggingface_hub
+type PyPointerFile = PyXetDownloadInfo;
+
 #[pyclass]
 #[derive(Clone, Debug)]
-pub struct PyPointerFile {
+pub struct PyXetDownloadInfo {
     #[pyo3(get, set)]
-    path: String,
+    destination_path: String,
```

Comment:
Wouldn't renaming the field in the struct/object break the backward compatibility too?

Reply:
This was handled by adding an extra getter/setter to the trait implemenation.

We've had to change this around to handle a backwards compatibility story through PyO3.